src/main/javascript/service/cat-api-service.js

```diff
@@ -323,6 +323,20 @@ function CatApiServiceProvider() {
             function $getCatApiService($http, catConversionService) {
             var catApiService = {};
 
+            /**
+             * This method allows to define (dynamic) endpoints after the configuration phase.
+             * @param {object} [settings] a new {EndpointConfig} will be created with the given settings
+             * @returns {CatApiEndpoint}
+             */
+            catApiService.$endpoint = function (settings) {
```

Comment:
please don't use $endpoint -> $ is used by angular and shouldn't be used for any custom names (https://docs.angularjs.org/api at the very top)

Note: the occurrences within cat-angular are legacy ones and will be removed in the future